File: frontend/src/api/traceabilityApi.ts
import api from './axios';
export const traceabilityApi = {
    getByBatchCode: (batchCode: string) => api.get(`/traceability/${batchCode}`),
    getAllLogs: () => api.get('/admin/traceability/logs'),
    verifyChain: () => api.get('/admin/traceability/verify'),
};
export const reviewApi = {
    create: (data: { product_id: string; rating: number; comment: string }) =>
        api.post('/reviews', data),
    getByProduct: (productId: string) => api.get(`/products/${productId}/reviews`),
};
export const userApi = {
    getAll: (params?: Record<string, string>) => api.get('/admin/users', { params }),
    verify: (id: string) => api.put(`/admin/users/${id}/verify`),
    updateStatus: (id: string, data: { is_verified: boolean }) =>
        api.put(`/admin/users/${id}/status`, data),
    create: (data: Record<string, any>) => api.post('/admin/users', data),
};
export const shrimpTypeApi = {
    getAll: () => api.get('/shrimp-types'),
    create: (data: { name: string; description?: string; image?: string }) =>
        api.post('/admin/shrimp-types', data),
    update: (id: string, data: Record<string, unknown>) => api.put(`/admin/shrimp-types/${id}`, data),
    delete: (id: string) => api.delete(`/admin/shrimp-types/${id}`),
};
export const dashboardApi = {
    getAdmin: () => api.get('/admin/dashboard'),
    getPetambak: () => api.get('/dashboard/petambak'),
    getLogistik: () => api.get('/dashboard/logistik'),
    getKonsumen: () => api.get('/dashboard/konsumen'),
};

File: fluttermultiplatform/lib/api/traceability_api.dart
import 'api_client.dart';

class TraceabilityApi {
  static Future getByBatchCode(String batchCode) => api.get('/traceability/$batchCode');
  static Future getAll() => api.get('/admin/traceability/logs');
  static Future verifyChain() => api.get('/admin/traceability/verify');
}

class ReviewApi {
  static Future create(Map<String, dynamic> data) => api.post('/reviews', data: data);
  static Future getByProduct(String productId) => api.get('/products/$productId/reviews');
}

class UserApi {
  static Future getAll({Map<String, dynamic>? params}) => api.get('/admin/users', queryParameters: params);
  static Future verify(String id) => api.put('/admin/users/$id/verify');
  static Future create(Map<String, dynamic> data) => api.post('/admin/users', data: data);
}

class ShrimpTypeApi {
  static Future getAll() => api.get('/shrimp-types');
  static Future create(Map<String, dynamic> data) => api.post('/admin/shrimp-types', data: data);
  static Future update(String id, Map<String, dynamic> data) => api.put('/admin/shrimp-types/$id', data: data);
  static Future delete(String id) => api.delete('/admin/shrimp-types/$id');
}

class DashboardApi {
  static Future getAdmin() => api.get('/admin/dashboard');
  static Future getPetambak() => api.get('/dashboard/petambak');
  static Future getLogistik() => api.get('/dashboard/logistik');
  static Future getKonsumen() => api.get('/dashboard/konsumen');
}
